### DISCUSSION TITLE
Halo: Combat Evolved VR - FAQ

### DISCUSSION THREAD

**Discussion by HoriZon-HZN** [LivingFray/HaloCEVR#66] [Q&A]
I have created a FAQ of some of the tips and tricks to improve the game more.

**[Halo: Combat Evolved VR - FAQ](https://seed-kilogram-837.notion.site/Halo-Combat-Evolved-VR-FAQ-15147cb726058074a296e35f1388bf0e?pvs=74)**

Thanks to Jayleigh, Americana Ninja for most of the info.

**Comment by slipperfish** [comment]
Thanks I've been trying to spread word on HaloCEVR but this makes it easy

> - **Can't Interact / Complete Tutorial Section**
>     
>     You need to use your mouse at certain points just wiggle it or open your ‘Halo’ folder
>     Rename d3d9.dll to d3d9aa.dll (this allows you to boot into flat screen mode)
>     Boot Halo and complete the entire tutorial and Save at Checkpoint after. Close game and reopen your ‘Halo’ folder
>     Rename d3d9aa.dll back to d3d9.dll (this allows you to boot back into VR mode)

I was able to switch to desktop view on the SteamVR dashboard and wiggle it with the controller/mouse cursor. Good because my PC is in another room.

**Comment by slipperfish** [comment]
Sorry should've read the guide slower the first time. Also asking about general readme/FAQ stuff here that applies to either page.

> RightHandMeleeSwingSpeed = 3.5 and LeftHandMeleeSwingSpeed = 3.5
> To make the melee reach out a little bit more. (Default is 2.5)

Make sure I'm not crazy but I thought this changed the speed threshold that would trigger a melee attack

> ShowRoomCentre = false
> Turn off the indicator at your feet to show where the player character is actually positioned.

I think this is a bad idea because you want to be standing on it. As far as I know this is your hitbox location and line-of-sight for enemies, if you don't stand in it you'll be cheating without realizing it. When moving you'll clip through map objects more because it's your collision point, and I haven't tested it but maybe you'll fall off ledges. You'll also be in the wrong spot for cutscenes.

> Make sure in game the video setting for v-sync is set to ‘NO VYSNC’

> chimera_vsync true
> Enables vertical sync to reduce screen tearing.

Asking anyone for expert commentary here. If this isn't a contradiction just wondered if we could get clarification that changing your vysnc setting one place but not another is correct and why. Also doesn't vsync cause input lag? Should you use vsync based on whether your framerate drops below your headsets refresh rate or not? Why do I not see screen tearing with vsync off? Am I blind?

> RECOMMNEDED CONSOLE COMMANDS FOR CHIMERA (use console with the tilde key "~")

I tried to run the console commands at the main menu and after loading a save, for most of them it told me I couldn't execute the commands. Example:
`Requested function "chimera_lod" cannot be executed right now.`

The latest readme informs me I could make a chimera_preferences.txt and point to it in chimera.ini. It runs the commands but tells me:
`Command not found`

The only one that worked in both methods was chimera_model_detail. I tried running in administrator and I use the default install of C:\Games\Halo\

**Reply by Schintilla** [reply]
You can mention that the Custom Edition installation exe is just a zip file and the maps can just be extracted without installation

**Comment by FirEmerald** [comment]
Here's a modified version of the FAQ, with some stuff removed or tweaked. Markdown format to allow it to potentially be added to the ReadMe, and includes instructions for stuff like enabling Custom Edition maps.
  
Resources: (Links are also provided in the instructions in case you miss one):
- [Halo Combat Evolved ISO](https://archive.org/details/HaloCEPC)
- [Halo PC (required) and CE (optional) 1.0.10 patches](https://archive.org/details/halopc-patch-1.0.10)
- [Halo CE VR mod](https://github.com/LivingFray/HaloCEVR/releases)
- [Chimera mod](https://github.com/SnowyMouse/chimera/releases) (optional, recommended, but required if using Custom Edition maps)
- [DirectSound to OpenAL wrapper](https://github.com/ThreeDeeJay/dsoal/releases/) (optional, recommended)
- [Halo PC/CE LAA (Long Adress Aware) patched EXEs](http://vaporeon.io/hosted/halo/mods/halo_laa_fix.7z) (optional, recommended)
- [Halo: Custom Edition](https://halomaps.org/hce/detail.cfm?fid=410) (optional, required for using Custom Edition maps)
- [Universal UI](https://halomaps.org/hce/detail.cfm?fid=6952) (optional, recommended when using Custom Edition maps)

Installing Halo: Combat Evolved:
- Mount [HALO.iso](https://archive.org/details/HaloCEPC) and run the setup file to install Halo: Combat Evolved
- - Halo CD key: `MHJYD-HCYX4-VC8TD-J32X8-7KD4B`
- - **IMPORTANT**: Install in a directory OTHER THAN Program Files (example: `C:\Games\Halo`).
- - - This is because `C:\Program Files` and `C\Program Files (x86)` are considered protected directories and **will** cause issues
- Run [halopc-patch-1.0.10.exe](https://archive.org/details/halopc-patch-1.0.10) to install the 1.0.10 patch

Installing the Halo CE VR Mod:
- Copy all the files from [HaloCEVR.zip](https://github.com/LivingFray/HaloCEVR/releases) into your Halo: Combat Evolved folder
- Run the game once to generate `VR/config.txt`
- Edit `VR/config.txt` (optional):
- - change `SnapTurn` to `false` to enable smooth turning
- - - change `SmoothTurnAmount` to `160`, as the default, `90`, is generally too slow
- - change `UnzoomedWeaponSmoothingAmount` to `0.2` to smooth out weapon movement
- - - change `Zoom1WeaponSmoothingAmount` to `0.5` to smooth out weapon movement more when zoomed in once
- - - change `Zoom2WeaponSmoothingAmount` to `0.8` to smooth out weapon movement even more when zoomed in twice
- - change `UIOverlayScale` to `12` to make the UI slightly larger and easier to see
- - change `ShowRoomCentre` to `false` to disable the indicator at the ground that shows where the player's position is considered to be (I recommend leaving it ON if you're playing standed in roomscale)
- - change `CrouchHeight` to adjust the height below which you will be considered to be "crouched" when physically crouching.
- - - set this to `0` to disable physical crouching
- - change `HandRelativeMovement` to 1 (or 2 for left-handed players) to change movement mode from view-oriented to controller-oriented
- - - change `HandRelativeOffsetRotation` to adjust the rotational offset of movement when using controller-oriented movement
- - change `ShowCrosshair` to `false` to disable the crosshair if you prefer
- - change `LeftHanded` to `true` to play in left-handed mode
- - - you should also swap the values of `LeftHandFlashlight` and `RightHandFlashlight`
- - - it is highly recommended to use the left-handed controller bindings from the game's SteamVR controller bindings page when using left-handed mode
- to disable and allow loading to desktop, rename `d3d9.dll` to `d3d9a.dll` - or remove it to uninstall.

Installing Chimera (Optional, recommended):
- rename `strings.dll` to `strings.old.dll`
- Copy `chimera.ini`, `strings.dll`, and `fonts` from [chimera-\<version\>.7z](https://github.com/SnowyMouse/chimera/releases) to the game folder
- Edit `chimera.ini`:
- - under `[font_override]`, change `enabled` to `0`
- Make sure in game the video setting for v-sync is set to ‘NO VYSNC’
- Recommended Console Commands (press `\`` to open the console from desktop view):
- - `chimera_model_detail 8` Change the model detail. Higher values increase LODs. This will not make models extra detailed. Instead, it will change the threshold for when higher LODs appear.
- - `chimera_block_mipmaps true` Prevents texture mipmaps from loading lower-resolution textures, ensuring sharper visuals.
- - `chimera_interpolation true` Smoothens animations by interpolating frames, making movements look more fluid.
- - `chimera_vsync true` Enables vertical sync to reduce screen tearing.
- To uninstall, simply remove the `strings.dll` that came with Chimera, and rename `strings.old.dll` back to `strings.dll`

Installing DSOAL for 3D Sound (Optional, recommended):
- Copy all the files from the `Win32` folder in [DSOAL+HRTF.zip](https://github.com/ThreeDeeJay/dsoal/releases/) to the game folder
- Edit `alsoft.ini`:
- - under `[general]` (create it if it doesn't exist) add or set `channels=stereo`, `frequency=48000`, `stereo-mode=headphones`, `cf_level=0`, `sources=512`, `sample-type=float32`, `hrtf=true`, `hrtf-mode=full`, and `period_size=960`
- - under `[reverb]` (create it if it doesn't exist) add `boost=-6`
- To uninstall, remove the `dsound.dll` file

Installing Halo LAA Fix (Increases the amount of RAM the game can use from 2GB to 3GB on 32-bit systems and 4GB on 64-bit systems) (Optional, recommended):
- Rename `halo.exe` inside your Halo folder to `halo.old.exe`
- Copy `retail/halo.exe` from [halo_laa_fix.7z](http://vaporeon.io/hosted/halo/mods/halo_laa_fix.7z)
- - You may or may not need to force the game to run as adminstrator to work now. Here's how to do it in case you need to:
- - - Right click `halo.exe`
- - - click on `Properties`
- - - go to the `Compatibility` tab
- - - make sure `Run this program as an administrator` is checked
- - If using Chimera, allow the game to load maps into RAM to potentially speed up loading (Optional):
- - - Edit `chimera.ini`:
- - - - Change line `;enable_map_memory_buffer=1` to `enable_map_memory_buffer=1` (remove the `;`)
- To uninstall (why?), remove the `halo.exe` from the LAA fix, and rename `halo.old.exe` back to `halo.exe`

Allow usage of Custom Edition maps:
- Follow the steps from `Installing Chimera` if you haven't already. Chimera is required for loading Custom Edition content.
- Install Halo: Custom Edition using [halocesetup_en_1.00.exe](https://halomaps.org/hce/detail.cfm?fid=410)
- - use the CD key from the `Installing Halo: Combat Evolved` section, and follow the same installation guidelines.
- - **DO NOT** install to the same folder as Halo! I Recommend installing to a folder beside it instead.
- Run [haloce-patch-1.0.10.exe](https://archive.org/details/halopc-patch-1.0.10) to install the 1.0.10 patch (might be optional unless you plan on using Custom Edition on it's own as well. It is currently NOT supported by the VR mod.)
- Copy and rename `bitmaps.map`, `loc.map`, and `sounds.map` from the Custom Edition `MAPS` folder to `custom_bitmaps.map`, `custom_loc.map`, and `custom_sounds.map` in the Combat Evolved `MAPS` folder
- Maps using custom player models and/or weapons may have graphical glitches

Installing Universal UI (optional but highly recommended to allow loading of Custom Edition singelplayer maps after following `Allow usage of Custom Edition maps`):
- rename `MAPS/ui.map` to `MAPS/ui.old.map`
- copy `MAPS/UI.map` from [Universal_UI_Version_1.1.zip](https://halomaps.org/hce/detail.cfm?fid=6952) to your `MAPS` folder
- To uninstall, remove the `MAPS/UI.map` from the Universal UI mod, and rename `MAPS/ui.old.map` to `MAPS/ui.map`.

Troubleshooting:
- When playing, if you are ever unable to interact with a menu or pop-up, ensure the game window has focus on your desktop.
- During the beginning of the original Halo campaign, you may have focus the game and wiggle the mouse (either physically or via the virtual cursor in SteamVR desktop view) to proceed past the part where you are asked to look around.
- Custom maps, including custom campaigns, may have player model issues - such as the hands not rotating with your view. In most cases, the weapons will still fire in the correct direction, but in some rare cases where the aiming has been modified, they will not. These cases are simply not compatible with VR.
- When playing the Lumoria campaign, the controls will be messed up during the start. Simply proceed over the first hill and they should automatically fix themselves.
- When experiencing any other issues, the first step should _always_ be to see if running as adminstrator fixes it. see `Installing Halo LAA Fix` for instructions on how to make the game _always_ run as adminstrator.
- The second step, if it's still not fixed, is to disable the VR mod and see if it's still broken. If so, your issue is not related to Halo CE VR, and you will need to find an alternative fix.
- Third, ensure the issue happens with original game content. Remove all modified content (apart from the VR mod, Chimera, and LAA), and try again. If it doesn't happen without them, you probably have an incompatible modification that you'll need to remove.
- Fourth, do a google search for your issue. See if someone else has had the same problem - they might have a fix.
- Finally, if none of the above steps have fixed your issue, you're probably safe to ask about it on the VR mod [discussions](https://github.com/LivingFray/HaloCEVR/discussions) or [issue tracker](https://github.com/LivingFray/HaloCEVR/issues)

Additional mods and content:
- [Halo Refined](https://proxeninc.net/Halo/Refined/V4Beta%20Download/halo_refined_retail_en_v4beta2.7z), a set of modified map files fixing bugs with the PC port and brining it closer to the original Xbox version
- - It will replace the vanilla campaign map files. Back them up if you don't want to lose them!
- [Halo: Movement Evolved](https://www.patreon.com/posts/halo-movement-v-117980524)
- - You will need to sign into your Patreon account (create one if you don't have one already) and then join the creator for free.
- - It will replace the vanilla campaign map files. Back them up if you don't want to lose them!
- - Incompatible with `Halo Refined`'s `sounds.map` and `ui.map`, you'll need be using the original files. You can download them from the same page as the mod. (unsure if Universal UI's ui.map works)
- - Notable features:
- - - Wall Running
- - - Sprint Mode
- - - No Fall Damage
- - - Improved Weapons
- - - First Person Warthogs
- - - And more! _Please_ read `WHAT ARE ALL THESE FOLDERS_README.txt` inside the zip file! There are somewhat significant changes you **will** want to know about!
- [Lumoria](https://halomaps.org/hce/detail.cfm?fid=6507), a custom campaign taking place onn the Forerunner world of Lumoria, starring a MK II Spartan named May.
- - If using Universal UI (recommended, see `Installing Universal UI`), do not install the ui.map file. Universal UI already supports the Lumoria campaign.
- - Highly recommend using Univeral UI (see `Installing Universal UI`) to load this campaign
- - Controls and view will be messed up during the first part of the campaign; proceed over the first hill and the controls will automatically be fixed.
- [Lumoria Refined](https://www.moddb.com/mods/lumoria-refined/downloads/lumoria-refined1), an improvement to the original Lumoria akin to how Halo Refined improves the original game
- - Replaces the original Lumoria maps. If using Universal UI, the original Lumoria is not required.
- You can find Custom Edition maps at [Halo CE maps](https://halomaps.org/hce/) but compatibility varies.

**Reply by FirEmerald** [reply]
Edit: Added troubleshooting, Halo: Movement Evolved, and uninstallation instructions. Cleaned up a bit.

**Reply by slipperfish** [reply]
Just a heads up but I was worried about `hrtf-mode=full` in alsoft.ini causing performance issues. I haven't had time to falsify test it but I was playing with it on and got CPU hickups in the room with a truckload of sleeping grunts in Assault on the Control Room, which should be a lot of audio sources playing at once.

**Reply by FirEmerald** [reply]
I just copied that from the original FAQ from the top, I didn't have any such issues but I do have a high-end gaming PC so that might be a factor.

**Reply by HoriZon-HZN** [reply]
ah cool will have to update it withy these new tips.

**Comment by slipperfish** [comment]
Github doesn't have direct messaging but I need your help @HoriZon-HZN. I stayed up two nights in a row trying to get dynamic lights/reflections to work. CEnshine is supposed to fix these as shown in this video. But it's Custom Edition only, and I can't get it to work at all while every other CEnshine effect works. It's frustrating because it's one of the best looking effects in the game and every plasma shot and grenade could look better:

https://www.youtube.com/watch?v=0egQ3MYwx7Y&t=311s

By coincidence I walk into this tutorial you made at one of those spots and the effect is there:

https://www.youtube.com/watch?v=rdlCu7IjbGI

But CEVR only works on retail, which can't use CEnshine. So what is your installation? What is the fix? 

Edit: So that room after the first bridge has a differently placed light that is stronger than the rooms I've been comparing it to, like the first room, so that's confusing me. But the game still looks better on your machine and I can't tell if it's placebo.

**Reply by HoriZon-HZN** [reply]
Not done anything different playing with gfx set to max and added the Chimera things from my guide.

**Comment by HoriZon-HZN** [comment]
OK updated the guide and gave you credit FirEmerald

**Reply by FirEmerald** [reply]
credit for what? almost all I posted came straight from your FAQ lol

**Reply by HoriZon-HZN** [reply]
Just a few bits and they way it was set out :)

**Comment by luci3dvr-hue** [comment]
Hi there I tried ninja tutorial and I managed to have halo vr work for original campaign works but when I try 4k+ hd texture or other custom campaign it crashes at launcher.
I use vr mod 1.4.0 and chimera v1.0.0,1200
Universal ui version 1.1
If I use original ui.map I can launch game but it crashes when I press new game or continue.

**Reply by Schintilla** [reply]
Can you provide a link to the 4k textures

**Reply by luci3dvr-hue** [reply]
Sure is one of extra campaigns:
https://www.moddb.com/games/halo-combat-evolved/addons/halo-ce-halo-4-mod-hd-textures-anniversary-mod-by-antrax

**Reply by Schintilla** [reply]
I tried the hd textures with VR 1.4, Chimera 1.0.0,1200 and Universal ui 1.1. Biggest problem was no weapons are visible. Gloves are there and you can fire but no gun. Did not check textures further but did not notice some odd things like ships clipping the ground. 

I did not check map loading. Previously it would always say I can download maps after selecting the difficulty level. No option to play. I did not check whether I actually had the maps even though they were listed.

**Reply by luci3dvr-hue** [reply]
Thank you for your research.
I will stick to original then :)

**Reply by Schintilla** [reply]
I tried the 4k textures again with the lastest fix to reinstate the weapons with Chimera 1215 and some success in also making some of the weapons appear with the 4k. At least later, the sniper rifle when first available was visible. Thereafter you could swap with other weapons but not all. So close but not quite there. Just need someone who understands this to look at it. Perhaps it is just a small tweak to make all the weapons appear.
